### PR TITLE
This is for displaying user profile

### PR DESCRIPTION
**Pull request by shechframgia**
This is for the task nos. 55696, 55695, 55712 and 55711

### CODE REVIEW

File: app/Http/Controllers/FollowController.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+use App\Models\User;
+use Illuminate\Http\Request;
+
+class FollowController extends Controller
+{
+    public function destroy(Request $request)
+    {
+        auth()->user()->followings()->detach($request->following_id);
+        return redirect('/user/' . auth()->user());
```
Comment (nguyenthanhtung88) [inline app/Http/Controllers/FollowController.php]: `auth()->user()` ???
You had `$this->user` from base controller right?

File: app/Http/Controllers/UserController.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+use App\Models\User;
+use Illuminate\Http\Request;
+
+class UserController extends Controller
+{
+    public function profile(User $user)
+    {
+        return view('user.show', compact('user'));
+    }
+    public function edit(User $user)
```
Comment (jsnulla) [inline app/Http/Controllers/UserController.php]: I think you need a space before this line

File: app/Http/routes.php [hunk 3]
```diff
@@ -28,18 +28,18 @@
...
-
     Route::group(['middleware' => 'auth'], function () {
+        Route::resourceparameters([
+            'categories' => 'category',
+            'words' => 'word',
+            'lessons' => 'lesson',
+        ]);
         Route::get('/home', 'HomeController@index');
-
+ 		Route::resource('user', 'UserController');
```
Comment (nguyenthanhtung88) [inline app/Http/routes.php]: Check indent

File: resources/views/user/show.blade.php [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+                    <div class="panel-heading">
+                        {{ trans('text.profile')}}
+                        @if(auth()->user()->id == $user->id)
+                            {{ link_to_action('UserController@edit', trans('text.edit'), $user->id, ['class' => 'btn btn-primary pull-right']) }}
+                        @endif
+                    </div>
+                    <div class="panel-body">
+                        <div class="row">
+                            <div class="col-md-3">
+                                {{ Html::image('/pictures/user_profile/' . $user->image, '', ['class' => 'media-object', 'data-holder-rendered' => 'true', 'width' => 100, 'height' => 100]) }}
```
Comment (nguyenthanhtung88) [inline resources/views/user/show.blade.php]: `/pictures/user_profile/` => this should be defined as config, don't fix text

File: resources/views/user/show.blade.php [hunk 5]
```diff
@@ -0,0 +1,59 @@
+@extends('layouts.app')
+
+@section('content')
+    <div class="container">
+        <div class="row">
+            <div class="col-md-8 col-md-offset-2">
+                <div class="panel panel-default">
+                    <div class="panel-heading">
+                        {{ trans('text.profile')}}
+                        @if(auth()->user()->id == $user->id)
```
Comment (nguyenthanhtung88) [inline resources/views/user/show.blade.php]: one space after `if`

File: resources/views/user/show.blade.php [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+                            <div class="col-md-9">
+                                {{ Form::label('name', trans('text.name'), ['class' => 'control-label']) }}
+                                {{ Form::label('name', $user->name, ['class' => 'form-control']) }}
+                                {{ Form::label('email', trans('text.email'), ['class' => 'control-label']) }}
+                                {{ Form::label('email', $user->email, ['class' => 'form-control']) }}
+                            </div>
+                        </div>
+                        <h3>{{ trans('text.following') }}</h3>
+                        <div class="col-md-12">
+                            @forelse($user->followings as $follow)
```
Comment (nguyenthanhtung88) [inline resources/views/user/show.blade.php]: one space before `forelse`

File: resources/views/user/show.blade.php [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+                                <div class="col-md-4">
+                                    <div class="media">
+                                        <div class="media-left">
+                                            {{ Html::image('/pictures/user_profile/' . $follow->image, '', ['class' => 'media-object', 'width' => 50, 'height' => 50 ]) }}
+                                        </div>
+                                        <div class="media-body">
+                                            <h4 class="media-heading">
+                                                {{ link_to('/user/' . $follow->id, $follow->name) }}
+                                            </h4>
+                                            @if(auth()->user()->id == $user->id)
```
Comment (nguyenthanhtung88) [inline resources/views/user/show.blade.php]: one space after `if`

File: app/Repositories/Eloquents/CategoryRepository.php [hunk 8]
```diff
@@ -10,7 +10,7 @@ public function createOrUpdateCategory($request, $category = null)
     {
         if (!empty($request->file('image_category'))) {
             $imageData = $request->file('image_category');
-            $path = public_path('pictures/category');
+            $path = public_path(Config::get('path.category'));
```
Comment (nguyenthanhtung88) [inline app/Repositories/Eloquents/CategoryRepository.php]: I think you should call `config()` helper function, it's shorter and we don't have to call statically :smiley:

**Comment by nguyenthanhtung88** [comment]
OK

**Pull request merged**